web/src/main/webapp/heatmap3d.ts

```diff
@@ -0,0 +1,804 @@
+import {
+    FullPage, IHtmlElement, Point, Size, KeyCodes,
+    significantDigits, formatNumber, translateString,
+    IScrollTarget, ScrollBar
+} from "./ui";
+import d3 = require('d3');
+import {RemoteObject, RemoteObjectView, Renderer, combineMenu, CombineOperators, SelectedObject, ZipReceiver} from "./rpc";
+import {ColumnDescription, Schema, ContentsKind, TableView, RecordOrder, TableRenderer, RangeInfo} from "./table";
+import {Pair, Triple, Converters, reorder, regression, ICancellable, PartialResult} from "./util";
+import {
+    BasicColStats, Histogram, ColumnAndRange, AnyScale, HistogramViewBase,
+    FilterDescription
+} from "./histogramBase";
+import {BaseType} from "d3-selection";
+import {ScaleLinear, ScaleTime} from "d3-scale";
+import {TopMenu, TopSubMenu} from "./menu";
+import {Histogram2DRenderer, Make2DHistogram} from "./histogram2d";
+
+// counterpart of Java class 'HeatMap3D'
+export class HeatMap3DData {
+    buckets: number[][][];
+    missingData: number;
+    totalsize: number;
+}
+
+class HeatMapLegend implements IHtmlElement {
+    private outer: HTMLElement;
+
+    // Fields for the axes.
+    private static axesTicks = 3;
+    private static readonly axesSize: Size = {
+        width: 200,
+        height: 200
+    };
+    private static readonly axesMargin = {
+        top: 40, right: 20, bottom: 40, left: 55
+    };
+
+    private axesG: any;
```

Comment:
For the `any` fields maybe you can add a comment indicating what they are supposed to be.